### PR TITLE
[WIP] add get_average_rydberg_densities to src/observables.jl

### PR DESCRIPTION
**Pull request by maolinml**
Adding a wrapper function `get_average_rydberg_densities` to src/observables.jl. The function returns the average Rydberg density throughout the evolution, determined by the given waveforms. 

Tests: Not done yet, but the function was tried on the lattice gauge theory example, which produces the correct result.

### CODE REVIEW

File: src/observables.jl [hunk 1]
```diff
@@ -37,3 +37,53 @@ rydberg_corr(reg) = rydberg_corr(Op.n, reg)
...
+
+    allwaveforms = [Ω, ϕ, Δ]
+    allwaveforms = allwaveforms[allwaveforms.!=nothing]
+    allwaveforms = reduce(vcat, allwaveforms)
+    duration = allwaveforms[1].duration
+    for i = 2 : length(allwaveforms)
+        if allwaveforms[i].duration != duration
+            error("The durations of waveforms are not consistent.")
+        end
+    end
```
Comment (Roger-luo) [inline src/observables.jl:76]: Can you explain what this does?

File: src/observables.jl [hunk 2]
```diff
@@ -37,3 +37,53 @@ rydberg_corr(reg) = rydberg_corr(Op.n, reg)
...
+
+    # Start the evolution 
+    h = rydberg_h(atoms, C=C, Δ=Δ, Ω=Ω, ϕ=ϕ)
+    prob = SchrodingerProblem(reg, duration, h, progress=true);
+    integrator = init(prob, Vern8());
```
Comment (Roger-luo) [inline src/observables.jl:81]: Can we make the Vern8 solver as a keyword?

**Review by Roger-luo — CHANGES_REQUESTED** [review]

File: src/observables.jl (hunk 2 above)
Comment (maolinml) [reply src/observables.jl]: Done, but could you specify the type of the `solver`? Thanks!

Comment (Roger-luo) [reply src/observables.jl]: I think we will leave the solver type here, since there are too many different kinds of solvers.

Comment (maolinml) [reply src/observables.jl]: Ok, then I will keep this file as it is now. I will add a few tests in the next couple of days.

**Review by Roger-luo — APPROVED** [review]
LGTM

**Pull request merged**